### ISSUE TITLE
Could `foldImageLayers` be added as lib function?

### ISSUE THREAD

**Issue by jashandeep-sohi** [nlewo/nix2container#108]
Appreciate the [pointer](https://blog.eigenvalue.net/2023-nix2container-everything-once/) in the readme on how to avoid duplicate files in layers. But could the `foldImageLayers` utility function mentioned in that post be included in this flake (or something similar)?

```nix
  foldImageLayers = let
    mergeToLayer = priorLayers: component:
      assert builtins.isList priorLayers;
      assert builtins.isAttrs component; let
        layer = nix2container.buildLayer (component
          // {
            layers = priorLayers;
          });
      in
        priorLayers ++ [layer];
  in
    layers: lib.foldl mergeToLayer [] layers;
```

I find myself duplicating it in multiple projects. Would be easier if it was available upstream.

**Comment by mikepurvis** [comment]
I agree that this would be great to add and document— as it stands, `buildLayer` is a bit of a footgun where you can easily end up re-shipping the same store paths if you're not careful to fold things together in the right way.

As a further enhancement, it would be nice if the layers were nameable/addressable somehow. Basically I have a project with multiple large images that intersect significantly with each other, and I want to define a hierarchy of shared layers upfront (say, A <- B <- C <- D <- E), and then have some of the images include all five, where others include just the first three or four.

**Comment by nlewo** [comment]
> But could the foldImageLayers utility function mentioned in that post be included in this flake 

Yes, a PR would be welcome ;)
